### PR TITLE
Added more detail to some error cases in the VertxHttpRecorder

### PR DESCRIPTION
**Pull request by GregJohnStewart**
There were some extremely generic and identical errors being thrown, making startup issues much harder to debug.

I added log statements and more descriptive errors.

If there is some detail I missed on why these were bind exceptions in stead of configuration exceptions, let me know and I can tweak back to something.

Additionally, some clarification would be great for the exact error cases and reasons in the L608 if block, to be more descriptive.

### CODE REVIEW

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 1]
```diff
@@ -147,6 +147,7 @@ public void handle(HttpServerRequest httpServerRequest) {
                 // we simply need to know if the URI is valid
                 new URI(httpServerRequest.uri());
             } catch (URISyntaxException e) {
+                LOGGER.error("Server request had bad URI.");
```
[the remark this answers is not in the document]
Comment (GregJohnStewart) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:150]: Yeah this seemed like an odd case to have to handle, but makes sense that one would.

Personally, I think there is merit to logging it, or even passing it back as part of the response, but I can just take this out if the group disagrees

**Comment by GregJohnStewart** [comment]
After some debugging, it is fairly purplexing to me, the logic in the L608 if block.... Debugging, the error at the start is an affirmative `java.net.BindException: Address already in use`, but then we go through the `if`'s, and deduce other errors?

**Comment by GregJohnStewart** [comment]
Ended up improving `QuarkusBindException` to give itself an appropriate message on the ports, so it is more obvious what's wrong.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:150]: I'm with @gsmet on this one

**Comment by GregJohnStewart** [comment]
Think this is ready if the group agrees

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 2]
```diff
@@ -619,15 +625,20 @@ public void handle(AsyncResult<String> event) {
...
+                                if ((sslConfig == null) && (httpServerOptions != null)) {
+                                    this.add(httpServerOptions.getPort());
+                                } else if ((httpConfiguration.insecureRequests == InsecureRequests.DISABLED) && (sslConfig != null)) {
+                                    this.add(sslConfig.getPort());
+                                } else if ((sslConfig != null) && (httpConfiguration.insecureRequests == InsecureRequests.ENABLED)
+                                           && (httpServerOptions != null)) {
+                                    this.addAll(List.of(httpServerOptions.getPort(), sslConfig.getPort()));
+                                }
+                            }}
+                        );
```
Comment (geoand) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:640]: Please avoid the double brace initialization pattern.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 3]
```diff
@@ -581,9 +581,15 @@ private void warnIfDeprecatedHttpConfigPropertiesPresent(HttpConfiguration httpC
...
-            Supplier<Integer> eventLoops, List<String> websocketSubProtocols, boolean auxiliaryApplication) throws IOException {
+    private static void doServerStart(
+            Vertx vertx,
+            HttpBuildTimeConfig httpBuildTimeConfig,
+            HttpConfiguration httpConfiguration,
+            LaunchMode launchMode,
+            Supplier<Integer> eventLoops,
+            List<String> websocketSubProtocols,
+            boolean auxiliaryApplication
+    ) throws IOException {
```
Comment (geoand) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:592]: Unless the formatter changed this on it's own, please revert this change

**Comment by geoand** [comment]
Aside from my comments on the code, please also rebase the PR onto the latest `main` and squash the commits.

**Comment by geoand** [comment]
> Finding it hard to do so, never squashed commits before, I'm squashing my commits? Apologies

No problem. If you are using IntelliJ, [this](https://youtu.be/ah9ENkAYDrY) shows how to do it easily

**Comment by geoand** [comment]
I've never seen that happen, so not sure what's up. You might need to do it manually with `git`

**Comment by cescoffier** [comment]
I just checked the commit, it looks good, but this needs to be rebased and there are some conflicts to resolve.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 4]
```diff
@@ -153,6 +153,7 @@ public class VertxHttpRecorder {
         @Override
         public void handle(HttpServerRequest httpServerRequest) {
             if (!uriValid(httpServerRequest)) {
+                LOGGER.error("Server request had bad URI.");
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:156]: As already mentioned, we need this one gone. We cannot log something when a URL is badly crafted, this is a security hazard.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 5]
```diff
@@ -670,15 +671,20 @@ public void handle(AsyncResult<String> event) {
...
+                            (BindException)effectiveCause,
+                            new ArrayList<>(){{
+                                if ((sslConfig == null) && (httpServerOptions != null)) {
+                                    this.add(httpServerOptions.getPort());
+                                } else if ((httpConfiguration.insecureRequests == InsecureRequests.DISABLED) && (sslConfig != null)) {
+                                    this.add(sslConfig.getPort());
+                                } else if ((sslConfig != null) && (httpConfiguration.insecureRequests == InsecureRequests.ENABLED)
+                                           && (httpServerOptions != null)) {
+                                    this.addAll(List.of(httpServerOptions.getPort(), sslConfig.getPort()));
+                                }
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:684]: I really don't like this pattern. Please just build a `List` before creating your `QuarkusBindException`, it will be simpler and more readable.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 6]
```diff
@@ -670,15 +671,20 @@ public void handle(AsyncResult<String> event) {
...
+                            }}
+                        );
+                        LOGGER.error(effectiveCause.getMessage());
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:687]: Are we sure we want this logged here? I thought the error was properly propagated to the user with the `completeExceptionally` below? /cc @cescoffier

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 7]
```diff
@@ -705,6 +710,7 @@ public void handle(AsyncResult<Void> event) {
                                     }
                                 });
                             } catch (Exception e) {
+                                LOGGER.warn("Failed to undeploy deployment.");
                                 e.printStackTrace();
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:714]: Instead of keeping the `e.printStackTrace();`, include `e` as the cause in the `warn()` call.

**Review by gsmet — CHANGES_REQUESTED** [review]
I fixed the PR to remove all the unnecessary commits and fixed the conflicts.

There are still a couple of things to fix though.

Could you have a look at my comments?

**Comment by GregJohnStewart** [comment]
Thought I fixed a couple of those.. on it

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java (hunk 5 above)
Comment (GregJohnStewart) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java]: After rebasing to my local, this looks out of date. Don;t know why this was seen as latest.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java (hunk 6 above)
Comment (GregJohnStewart) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java]: I threw it in as a completeness thing, if we are sure it is reported then it's okay

**Comment by cescoffier** [comment]
Can you rebase and make sure you produce a single commit?

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 8]
```diff
@@ -705,7 +708,7 @@ public void handle(AsyncResult<Void> event) {
                                     }
                                 });
                             } catch (Exception e) {
-                                e.printStackTrace();
+                                LOGGER.warn("Failed to undeploy deployment: ", e);
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:711]:
```suggestion
                                LOGGER.warn("Failed to undeploy deployment ", e);
```

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 9]
```diff
@@ -670,24 +670,27 @@ public void handle(AsyncResult<String> event) {
...
-                                && (httpServerOptions != null)) {
-                            effectiveCause = new QuarkusBindException(
-                                    List.of(httpServerOptions.getPort(), sslConfig.getPort()));
+                                   && (httpServerOptions != null)) {
+                            portsUsed = List.of(httpServerOptions.getPort(), sslConfig.getPort());
                         }
+                        
+                        effectiveCause = new QuarkusBindException((BindException)effectiveCause, portsUsed);
                     }
+                    LOGGER.error(effectiveCause.getMessage());
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java:686]: Yeah, I think we should drop it. I have already seen this error message in the console so I think it's properly propagated. It's easy to check it though if you want, just start the server with an already used port.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a few suggestions, we're nearly there. Thanks for your patience!

**Comment by GregJohnStewart** [comment]
Should be good to go now, perhaps pending another squash

**Comment by gsmet** [comment]
I rebased, fixed the conflicts and fixed the formatting. I'll check what CI has to say tomorrow.

**Review by gsmet — APPROVED** [review]
